scripts/hypothesistooling.py

```diff
@@ -146,3 +146,28 @@ def build_jobs():
         status = m['state']
         jobs.setdefault(status, []).append(name)
     return jobs
+
+
+def modified_files():
+    files = set()
+    for command in [
+        ['git', 'diff', '--name-status', latest_version(), 'HEAD'],
+        ['git', 'diff', '--name-status']
+    ]:
+        diff_output = subprocess.check_output(command).decode('ascii')
+        for l in diff_output.split('\n'):
+            if l:
+                parts = l.split('\t')
+                status = parts[0]
+                file = parts[-1]
+                assert status in ('M', 'A', 'D'), status
```

Comment:
This may fail, because `git diff` can have plenty of other codes. From the `git diff` manpage:

> `--diff-filter=[ACDMRTUXB*]`
>
> Select only files that are Added (A), Copied (C), Deleted (D), Modified (M), Renamed (R), have their type (i.e. regular file, symlink, submodule, ...) changed (T), are Unmerged (U), are Unknown (X), or have had their pairing Broken (B). Any combination of the filter characters may be used. When * (All-or-none) is added to the combination, all paths are selected if there is any file that matches other criteria in the comparison; if there is no file that matches other criteria, nothing is selected.

Better might be to use a command something like:

```console
git diff --name-only --diff-filter=d
```

which gets the name of all files which don't have a "deleted" status.